### DISCUSSION TITLE
Implementing User Authentication in llama.cpp Server

### DISCUSSION THREAD

**Discussion by SeanZhang7** [ggml-org/llama.cpp#7485] [Q&A]
I am working on modifying index.html in llama.cpp/examples/server/public to add a user login window. This login window will prompt users to input their username and password to access the server. I want to manage user information in a .js file, allowing new users to be added without having to rebuild the entire project.

Methods I've Tried:
Creating a users.js file in public and importing it in index.html
Result: This leads to a 404 error.
Creating a users.js file in public, importing it in index.html, adding it to the Makefile to generate users.js.hpp, then include and Get users.js.hpp in server.cpp
Result: This requires a rebuild every time a new user is added.
Question:
Does anyone have a better approach for achieving this?

**Comment by marcingomulkiewicz** [comment]
Proxy llama server with Apache or anything? :slightly_smiling_face:

**Reply by SeanZhang7** [reply]
Thank you for your response. This project is an informal experiment where I am trying to build a user login system on my own. It doesn't need to be complex. Could you kindly suggest any simpler approaches?

**Comment by SeanZhang7** [comment]
I solved this by storing usernames and passwords in local storage and encrypt them with SHA-256.
